README.md

```diff
@@ -29,28 +42,133 @@ running `sbcl` and evaluating `ql:*local-project-directories*`. Once
+priority order.
+
+One can be even finer-grained than `with-backends`. Given a function
+`f` which has many backend implementations, one can get a specific
+implementation by using the function:
+
+```
+(magicl.backends:backend-implementation 'f :backend-name)
+```
+
+For instance
+
+```
+(magicl.backends:backend-implementation 'magicl:csd :lapack)
+```
+
+will give the implementation of the cosine-sine decomposition function
+in LAPACK. This can be called in exactly the same way `magicl:csd` can
+be called.
+
+In `backend-implementation`, if both the function name and the backend
+name are (quoted) constants, this will be looked up at compile-time,
+which is useful for writing efficient code that does not dispatch. But
+note that by doing this, `with-backends` will not be respected.
+
 ## Testing MAGICL
 
 You can run the MAGICL tests from your Lisp REPL with:
 
 ```
-(ql:quickload :magicl-tests)
 (asdf:test-system :magicl)
 ```
 
+You currently need all of the extensions working for the tests to run.
+
 ## High-level Interface
 
-See [high-level doc](doc/high-level.md).
+See the [high-level doc](doc/high-level.md) for an extensive discussion
+and comparison of MAGICL functions with those of MATLAB and NumPy.
+
+## Developer's Guide: How to Add New Functions
+
+See the [developer how-to](doc/dev-how-to.md) to understand how to add
+new functionality to MAGICL.
 
 ## Showing Available Functions
 
-Some distributions of a library don't actually provide all of the functions of the reference BLAS and LAPACK. One can look at a summary of available and unavailable functions with the function `magicl:print-availability-report`. By default, it will show all functions and their availability. There are three arguments to fine-tune this behavior:
+Some distributions of a library don't actually provide all of the
+functions of the reference BLAS and LAPACK. One can look at a summary
+of available and unavailable functions with the function
+`magicl:print-availability-report`. By default, it will show all
+functions and their availability. There are three arguments to
+fine-tune this behavior:
+
+1. Key `:show-available <boolean>` (default `t`): show available
+functions
+
+2. Key `:show-unavailable <boolean>` (default `t`): show unavailable
+functions
 
-1. Key `:show-available <boolean>` (default `t`): show available functions
-2. Key `:show-unavailable <boolean>` (default `t`): show unavailable functions
-3. Key `:search <string>`: only show functions which have `<string>` as a substring. **This argument takes into account the previous two arguments.**
+3. Key `:search <string>`: only show functions which have `<string>`
+as a substring. **This argument takes into account the previous two
+arguments.**
 
-For example, we can look for all available functions which might relate to `svd` by doing the following:
+For example, we can look for all available functions which might
+relate to `svd` by doing the following:
```

Comment:
This does not seem like a very good example as things currently stand, because the current listing is all using Fortran names

```
CL-USER> (magicl.foreign-libraries:print-availability-report)
        Fortran Function        Lisp Function
------------------------------------------------------------------------

Library LIBEXPOKIT: /Users/erik/.quicklisp/local-projects/magicl/expokit/libexpokit.dylib
    [x] DGCHBV                  MAGICL.EXPOKIT-CFFI:%DGCHBV
    [x] DGPADM                  MAGICL.EXPOKIT-CFFI:%DGPADM
    [x] DNCHBV                  MAGICL.EXPOKIT-CFFI:%DNCHBV
    [x] DSCHBV                  MAGICL.EXPOKIT-CFFI:%DSCHBV
    [x] DSPADM                  MAGICL.EXPOKIT-CFFI:%DSPADM
    [x] ZGCHBV                  MAGICL.EXPOKIT-CFFI:%ZGCHBV
    [x] ZGPADM                  MAGICL.EXPOKIT-CFFI:%ZGPADM
    [x] ZHPADM                  MAGICL.EXPOKIT-CFFI:%ZHPADM
    [x] ZNCHBV                  MAGICL.EXPOKIT-CFFI:%ZNCHBV
```

Reply:
To elaborate, I just mean that there's nothing about SVD in the report.

```
CL-USER> (magicl.foreign-libraries:print-availability-report :search "svd")
        Fortran Function        Lisp Function
------------------------------------------------------------------------

Library LIBEXPOKIT: /Users/erik/.quicklisp/local-projects/magicl/expokit/libexpokit.dylib
```

Reply:
I agree, it's kind of hokey. I'm going to shuffle this elsewhere.

Reply:
Actually, this uncovered a series of bugs in the MAGICL-GEN system. I fixed them, so now the SVD search should work.